core/src/main/java/org/infinispan/marshall/core/GlobalMarshaller.java

```diff
@@ -567,12 +626,13 @@ private void writeFlagsWithExternalizer(BytesObjectOutput out, Class<?> componen
       }
    }
 
-   void writeUnknownClean(Marshaller marshaller, Object obj, ObjectOutput out) {
-      try {
-         writeUnknown(marshaller, obj, out);
-      } catch (IOException e) {
-         throw new CacheException(e);
-      }
+   private boolean isLambda(Class<?> clazz, Object obj) {
+      return clazz.isSynthetic() || obj instanceof SerializableFunction;
```

Comment:
Hrmm, what is the check for `SerializableFunction` here for? Is this the only serializable lambda we detect? Sadly there are a lot more :(

https://github.com/infinispan/infinispan/tree/master/core/src/main/java/org/infinispan/util/function

Reply:
I think the `isSynthetic` check is sufficient, as `instanceOf` those interfaces does not necessarily mean that it's a lambda, in which case we should not be using the LambdaMarshaller.

Reply:
That is what I was thinking in regards to syntheic.